src/Utilities/Compiler/RuleLevel.cs

```diff
@@ -11,9 +11,15 @@ internal enum RuleLevel
 #pragma warning restore CA1008 // Enums should have zero value
     {
         /// <summary>
-        /// Correctness rule which should have <b>no false positives</b>, and is required to be fixed by users.
-        /// This rule will be <b>enabled in CI and IDE live analysis</b> by default with severity <see cref="DiagnosticSeverity.Error"/>.
+        /// Correctness rule which prevents the compiler from producing a well-defined output binary, and must have
+        /// <b>no false positives</b>. Violations of this rule must be fixed by users before any other testing work can
+        /// continue. This rule will be <b>enabled in CI and IDE live analysis</b> by default with severity
+        /// <see cref="DiagnosticSeverity.Error"/>.
         /// </summary>
+        /// <remarks>
+        /// Since analyzers cannot directly influence output binaries, this value is only valid in the implementation of
```

Comment:
Only thing I'd note here is that the `DetectPreviewFeatures` analyzer will end up using `BuildError` (it is not a source generator) eventually. I don't think we need to call it out here or make any changes to the wording. As long as everyone is aware of it I'm fine with the wording here.

Reply:
This section was specifically intended to clarify that `DetectPreviewFeaturesAnalyzer` will use `BuildWarning`, not `BuildError`.

Reply:
> This section was specifically intended to clarify that `DetectPreviewFeaturesAnalyzer` will use `BuildWarning`, not `BuildError`.

@sharwell In that case, should it still report diagnostics for generated code?

Reply:
There is no direct impact here on decisions to analyze and/or report diagnostics on generated code.